aten/src/ATen/TensorGeometry.h

```diff
@@ -113,6 +113,27 @@ struct TORCH_API TensorGeometry {
     return r;
   }
 
+  std::vector<c10::SymInt>& mutable_sizes() {
```

Comment:
What's the thread-safety story on these?
I haven't dived into how they are used in the other PR but this structure is saved on the graph which can be used in multiple threads at the same time. So if this is not read-only anymore, do we need to add locking?

Reply:
The code using this in the other PR only uses a single thread.

Reply:
How do you guarantee that? The user could be calling backward() from multiple threads?

Reply:
We could add a lock around all of compiled_autograd.  Since we are mutating nodes in-place, it is not thread safe.

Reply:
Actually, this isn't needed.  This is already protected by:
https://github.com/pytorch/pytorch/blob/548ad77367b5fab93062db4284b1f3af83822158/torch/csrc/dynamo/compiled_autograd_py.cpp#L285

Reply:
The gil doesn't create critical blocks. In can be released in most places in there and another thread pick up.
Also you could have a vanilla backward running in another thread that would not hold the gil right?

Reply:
I added a lock in the other PR.  compiled_autograd is enabled per-process, so mixing it between threads is not supported.